runtime/doc/options.txt

```diff
@@ -2217,21 +2217,16 @@ A jump table for the options with a short description can be found at |Q_op|.
 	viminfo file, etc.  It sets the kind of characters which Vim can work
 	with.  See |encoding-names| for the possible values.
 
-	NOTE: Changing this option will not change the encoding of the
-	existing text in Vim.  It may cause non-ASCII text to become invalid.
-	It should normally be kept at its default value, or set when Vim
-	starts up.  See |multibyte|.  To reload the menus see |:menutrans|.
-
-	This option cannot be set from a |modeline|.  It would most likely
-	corrupt the text.
-
-	NOTE: For GTK+ 2 it is highly recommended to set 'encoding' to
-	"utf-8".  Although care has been taken to allow different values of
-	'encoding', "utf-8" is the natural choice for the environment and
-	avoids unnecessary conversion overhead.  "utf-8" has not been made
-	the default to prevent different behavior of the GUI and terminal
-	versions, and to avoid changing the encoding of newly created files
-	without your knowledge (in case 'fileencodings' is empty).
+	This option cannot be changed after sourcing of |vimrc| files is done.
+	Otherwise it would cause non-ASCII text inside Vim to become
+	invalid. It should normally be kept at its default value, or be set
+	in vimrc. See |multibyte|. To reload the menus see |:menutrans|.
+
+	NOTE: When using Nvim with a GUI or remote plugins it is highly
+	recommended to set 'encoding' to "utf-8". Plugins are free to
+	assume that 'encoding' is "utf-8" and might only support non-ASCII
+	text in this case. On most modern platforms the default value will
+	already be "utf-8".
```

Comment:
Paragraph can be replaced with:

    The recommended 'encoding' is "utf-8". Remote plugins only support utf-8.